sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosDataWriteFactory.scala

```diff
@@ -24,18 +24,10 @@ class CosmosDataWriteFactory(userConfig: Map[String, String],
   class CosmosWriter(inputSchema: StructType) extends DataWriter[InternalRow] {
     logInfo(s"Instantiated ${this.getClass.getSimpleName}")
 
-    val cosmosAccountConfig = CosmosAccountConfig.parseCosmosAccountConfig(userConfig)
     val cosmosTargetContainerConfig = CosmosContainerConfig.parseCosmosContainerConfig(userConfig)
     val cosmosWriteConfig = CosmosWriteConfig.parseWriteConfig(userConfig)
 
-    // TODO moderakh: this needs to be shared to avoid creating multiple clients
-    val builder = new CosmosClientBuilder()
-      .key(cosmosAccountConfig.key)
-      .endpoint(cosmosAccountConfig.endpoint)
-      .consistencyLevel(ConsistencyLevel.EVENTUAL)
-
-    SparkBridgeInternal.setMetadataCacheSnapshot(builder, cosmosClientStateHandle.value)
-    val client = builder.buildAsyncClient();
+    val client = CosmosClientCache(CosmosClientConfiguration(userConfig, useEventualConsistency = true), Some(cosmosClientStateHandle))
```

Comment:
CosmosConfigEntry has infra for supporting default value, could you please rely on that instead of passing the default explicitly here?

see sample please:
https://github.com/moderakh/azure-sdk-for-java/blob/eb13c04777dd8fb183f30e4b217e6c3bba573dcc/sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala#L102

Reply:
But this is a Write flow, and the config is for Read operations.